### DISCUSSION TITLE
How would I handle local storage quota reached errors using `persist`?

### DISCUSSION THREAD

**Discussion by phultquist** [pmndrs/zustand#1938] [Q&A]
Hi there,

We've got a zustand setup that is roughly:

```tsx
const useMyState = create<MyType>()(
  persist(
    (set, get): MyType => ({
      foo: (prop) => (newValue) => set({ [prop]: newValue }),
    }),
    {
      name: 'my-ls-key',
      partialize(state) {
        for (const key of LOCALSTORAGE_BLOCKLIST) {
          if (key in state) delete state[key];
        }
        return state;
      },
      version: 1,
    }
  )
);
```

Some of our users are getting errors saving to local storage due to a quota exceeded. These errors crash our app and are handled by the error boundary. I'm wondering if there's a way I can handle these errors so they don't throw? I briefly looked in the docs and couldn't find anything.Is wrapping the `persist()` in a `try/catch` sufficient? 

Thanks!

Specific error: 
```
Failed to execute 'setItem' on 'Storage': Setting the value of 'ls-key' exceeded the quota.
```

**Comment by JClackett** [comment]
yeah, really need a solution for this too, any ideas?

Also is there a way to customise what happens if JSON.parse fails?

**Reply by dbritto-dev** [reply]
@JClackett to handle that you can use a custom storage when you can add some checks

**Comment by ofk** [comment]
I have faced this issue too.

I am using zustand/persist to write a lot of data to localStorage. The data includes a JSON string `updated_at`. When `DOMException: The quota has been exceeded` happens, I want to delete old data from localStorage and try again.

Is there any workaround for onError or retry?

**Reply by dbritto-dev** [reply]
@ofk you can implement that with custom storage
